test5.rb

```diff
@@ -0,0 +1,136 @@
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/CyclomaticComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    c = 0
+    while c < to_a.length
+      yield(to_a[c])
+      c += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    c = 0
+    while c < to_a.length
+      yield(to_a[c], c)
+      c += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_each { |item| result << item if yield(item) }
+    result
+  end
+
+  def my_all?(pattern = nil)
+    c = 0
+    statement = true
+    if block_given?
+      to_a.length.times do
+        statement = false unless yield(to_a[c])
+        c += 1
+      end
+    elsif !pattern.nil?
+      to_a.length.times do
+        statement = false if to_a[i].scan(pattern)
+      end
+    end
+    statement
+  end
+
+  def my_none?(pattern = nil)
+    c = 0
+    statement = true
+    if block_given?
+      to_a.length.times do
+        statement = false if yield(to_a[c])
+        c += 1
+      end
+    elsif !pattern.nil?
+      to_a.length.times do
+        statement = false if to_a[i].scan(pattern)
+      end
+    end
+    statement
+  end
```

Comment:
### #my_none? should behave as follow:
- [ ] 17. when no block or argument is given returns false only if one of the collection members is true
- [ ] 18. when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] 19. when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
- [ ] 20. when a class is passed as an argument returns false if any of the collection is a member of such class
- [ ] 21. when a Regex is passed as an argument returns true if none of the collection matches the Regex
- [ ] 22. when a Regex is passed as an argument returns false if any of the collection matches the Regex
- [ ] 23. when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
- [ ] 24. when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern